roles/openshift_health_checker/openshift_checks/diskspace_availability.py

```diff
@@ -0,0 +1,94 @@
+
+    def run(self, tmp, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        group_names = get_var(task_vars, "group_names", default=[])
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        diskfree = self.to_gibibytes(self.get_disk_space(ansible_mounts))
+
+        # if running containerized installation check
+        # that available space is not below 5% instead
+        if is_containerized:
+            docker_facts = self.module_executor("docker_facts", {}, task_vars)
+            info = docker_facts.get("docker_facts")
+            status = dict(info.get("DriverStatus"))
+
+            if info.get("Driver", None) == self.supported_docker_dev_driver:
+                disktotal = status.get("Data Space Total", None)
+                diskavail = status.get("Data Space Available", None)
+                diskleft_percent = self.to_bytes(diskavail) / self.to_bytes(disktotal) * 100
+
+                # fail if less than 5% of available dataspace space left
+                if int(diskleft_percent) < self.minimum_docker_space_percent:
+                    msg = "The amount of data space remaining (%s%%) for the docker storage driver is below %s%%" \
+                          % (int(diskleft_percent), self.minimum_docker_space_percent)
+                    return {"failed": True, "msg": msg}
+
+            return {"changed": False}
+
+        recommended_diskspace_gb = self.recommended_master_space_gb
+
+        if "nodes" in group_names:
+            recommended_diskspace_gb = self.recommended_node_space_gb
+
+        if float(diskfree) < recommended_diskspace_gb:
+            msg = "Available disk space for %s (%s) below recommended storage. Minimum required disk space: %s GB" \
+                  % ("node" if "nodes" in group_names else "master", str(diskfree) + " GB", recommended_diskspace_gb)
+            return {"failed": True, "msg": msg}
+
+        return {"changed": False}
+
+    @staticmethod
+    def get_disk_space(ansible_mounts):
+        if not ansible_mounts:
+            return None
+
+        root_size_available = None
+        for mnt in ansible_mounts:
+            if mnt.get("mount") == "/":
+                root_size_available = mnt.get("size_available")
+            if mnt.get("mount") == "/var":
+                return mnt.get("size_available")
+
+        return root_size_available
+
+    @staticmethod
+    def to_gibibytes(total_bytes):
+        return total_bytes / 1073741824
+
+    @staticmethod
+    def to_bytes(strfmt):
+        units = {
+            "b": 1,
+            "kib": 1024,
+            "kb": 1000,
+            "mib": 1024 ** 2,
+            "mb": 1000 ** 2,
+            "gib": 1024 ** 3,
+            "gb": 1000 ** 3,
+            "tib": 1024 ** 4,
+            "tb": 1000 ** 4,
+            "pib": 1024 ** 5,
+            "pb": 1000 ** 5
+        }
+
+        segments = strfmt.lower().split(" ")
```

Comment:
There can easily be an `AttributeError` exception here. We use `strfmt` as if it was a `str`, but in the code above we may clearly be calling this function with `None` as the first argument...

In a dynamically typed language, we have to keep our usage of data types sane.